src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v1/mongo/DeadLetterDataProvider.kt

```diff
@@ -45,21 +45,30 @@ class DeadLetterDataProvider(
     override fun totalRecordCount(
         searchParams: EcommerceSearchDeadLetterEventsRequestDto
     ): Mono<Int> {
-        val source = searchParams.source!!
+        val source = searchParams.source ?: return Mono.just(0)
         val timeRange = searchParams.timeRange
+        val excludeStatuses = searchParams.excludeStatuses
+
+        val startDate = timeRange?.startDate?.toString() ?: ""
+        val endDate = timeRange?.endDate?.toString() ?: ""
+        val npgStatuses = excludeStatuses?.npgStatuses ?: emptyList()
+        val eCommerceStatuses = excludeStatuses?.ecommerceStatuses ?: emptyList()
+
         return when (source) {
             DeadLetterSearchEventSourceDto.ALL -> {
                 if (timeRange != null) {
-                    val startDate = timeRange.startDate.toString()
-                    val endDate = timeRange.endDate.toString()
                     logger.info(
-                        "Counting all dead letter events in time range {} - {}",
+                        "Counting all dead letter events in time range {} - {}, with eCommerceStatus not in {} and npgStatus not in {}",
                         startDate,
-                        endDate
+                        endDate,
+                        npgStatuses.toString(),
+                        eCommerceStatuses.toString(),
```

Comment:
```suggestion
                        npgStatuses,
                        eCommerceStatuses,
```
remove explicit toString invocation here. input value conversion to string is automatically performed by logging framework in a null-safe manner

Reply:
Solved with https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/pull/185/commits/5a6ff7e50f84eda764802f0759de526d866fbe87